atomic_reactor/plugins/post_group_manifests.py

```diff
@@ -0,0 +1,211 @@
+
+from six.moves.urllib.parse import urlparse
+
+from atomic_reactor.plugin import PostBuildPlugin, PluginFailedException
+from atomic_reactor.util import Dockercfg, get_manifest_digests
+
+
+class GroupManifestsPlugin(PostBuildPlugin):
+    key = 'group_manifests'
+    is_allowed_to_fail = False
+
+    def __init__(self, tasker, workflow, registries, group=True, goarch=None):
+        """
+        constructor
+
+        :param tasker: DockerTasker instance
+        :param workflow: DockerBuildWorkflow instance
+        :param registries: dict, keys are docker registries, values are dicts containing
+                           per-registry parameters.
+                           Params:
+                            * "secret" optional string - path to the secret, which stores
+                              login and password for remote registry
+        :param group: bool, if true, create a manifest list; otherwise only add tags to
+                      amd64 image manifest
+        :param goarch: dict, keys are platform, values are go language platform names
+        """
+        # call parent constructor
+        super(GroupManifestsPlugin, self).__init__(tasker, workflow)
+        self.group = group
+        self.goarch = goarch or {}
+        self.registries = registries
+        self.worker_registries = {}
+
+    def submit_manifest_list(self, registry, registry_conf, manifest_list_spec):
+        docker_push_secret = registry_conf.get('secret', None)
+        with NamedTemporaryFile(prefix='manifest-list', suffix=".yml", mode='w') as fp:
+            yaml.dump(manifest_list_spec, stream=fp)
+            fp.flush()
+            self.log.info("Wrote to file %s", fp.name)
+
+            cmd = ['manifest-tool', '--docker-cfg=%s' % docker_push_secret,
+                   'push', 'from-spec', fp.name]
+            subprocess.check_call(cmd)
+            self.log.info("Manifest list submitted")
+
+    def get_grouped_manifests(self):
+        grouped_manifests = []
+        for registry, registry_conf in self.registries.items():
+            manifest_list_spec = {}
+            manifest_list_spec['manifests'] = []
+            all_annotations = self.workflow.build_result.annotations['worker-builds']
+            for platform in all_annotations:
+                worker_image = all_annotations[platform]['digests'][0]
+                digest = worker_image['digest']
+                repository = worker_image['repository']
+                arch_entry = {
+                    'image': '{0}/{1}@{2}'.format(registry, repository, digest),
+                    'platform': {
+                        'os': 'linux',
+                        'architecture': self.goarch.get(platform, platform)
+                    }
+                }
+                manifest_list_spec['manifests'].append(arch_entry)
+
+            image_tags = []
+            for image in self.workflow.tag_conf.images:
+                image_tags.append(image.to_str(registry=False))
```

Comment:
We want just the suffix part here I think, not the repository name. Also, we only need to set additional tags other than the one used in 'image'.
```
registry_image = self.workflow.tag_conf.images[0].copy()
image_tags = [image.tag for image in self.workflow.tag_conf.images[1:]]
```